### DISCUSSION TITLE
yaml parser

### DISCUSSION THREAD

**Discussion by raphael-grimm** [nlohmann/json#3883] [Ideas]
I am working with yaml and json config files. Currently I am converting the yaml data into json data for processing since the API for nlohmann::json is better to work with. I was thinking about implementing a yaml (de)serializer using nlohmann::json as the backend datastructure. 
I could implement this as a standalone project or add the parser to this codebase.
Would a yaml parser be seen as out of scope for this codebase?

**Comment by gregmarr** [comment]
Seems like that ability would fit with these other abilities https://github.com/nlohmann/json#binary-formats-bson-cbor-messagepack-ubjson-and-bjdata, but it's up to @nlohmann to make that determination.

**Reply by nlohmann** [reply]
As much as I would like to see more features like this, I am afraid I will not be able to maintain it in the long run:

- YAML is so much more complex than JSON (or the binary formats in this library), that it will be a constant inflow of feature requests, bugs, etc.
- There are already so many open ends that maintaining the status quo is close to impossible. Adding more to the core of the library would worsen the situation.

**Reply by gregmarr** [reply]
Yeah, that makes sense.

**Reply by LecrisUT** [reply]
One thing to consider is that with tools like https://github.com/pboettch/json-schema-validator it would become easier to design the validator if they had a common interface instead of designing a custom one for whichever yaml parser could be used tehre.

**Reply by acampove** [reply]
> As much as I would like to see more features like this, I am afraid I will not be able to maintain it in the long run:
> 
>     * YAML is so much more complex than JSON (or the binary formats in this library), that it will be a constant inflow of feature requests, bugs, etc.
> 
>     * There are already so many open ends that maintaining the status quo is close to impossible. Adding more to the core of the library would worsen the situation.

I keep hearing that _any system that can work with JSON should work with YAML_ If that is true, how is this package not able to read YAML too?

**Reply by nlohmann** [reply]
No, it's the other way around: YAML is a strict superset of JSON.
